src/pytest_codspeed/plugin.py

```diff
@@ -266,11 +238,17 @@ def __call__(self, func: Callable[..., T], *args: Any, **kwargs: Any) -> T:
         config = self._request.config
         plugin = get_plugin(config)
         plugin.benchmark_count += 1
-        if plugin.is_codspeed_enabled and plugin.should_measure:
-            assert plugin.lib is not None
-            return _run_with_instrumentation(
-                plugin.lib, self._request.node.nodeid, config, func, *args, **kwargs
-            )
+        if (
+            plugin.is_codspeed_enabled
+            and plugin.lib is not None
+            and plugin.should_measure
+        ):
+            return wrap_pyfunc_with_instrumentation(
+                plugin.lib,
+                self._request.node.nodeid,
+                config,
+                func,
+            )(*args, **kwargs)
```

Comment:
Don't hesitate to add that as an additional test

Reply:
I started to get really confused with the intention here since the above example also fails with `master`, see https://github.com/CodSpeedHQ/pytest-codspeed/pull/30

I suspect we need to do something similar to what `pytest-rerunfailures` does to clear the cached results: https://github.com/pytest-dev/pytest-rerunfailures/blob/a53b9344c0d7a491a3cc53d91c7319696651d21b/src/pytest_rerunfailures.py#L565-L567